apollo-normalized-cache/src/main/java/com/apollographql/apollo/cache/normalized/internal/NoOpApolloStore.kt

```diff
@@ -0,0 +1,133 @@
+  }
+
+  override fun remove(cacheKeys: List<CacheKey>): ApolloStoreOperation<Int> {
+    return emptyOperation(0)
+  }
+
+  override fun networkResponseNormalizer(): ResponseNormalizer<Map<String, Any>> {
+    return ResponseNormalizer.NO_OP_NORMALIZER as ResponseNormalizer<Map<String, Any>>
+  }
+
+  override fun cacheResponseNormalizer(): ResponseNormalizer<Record> {
+    return ResponseNormalizer.NO_OP_NORMALIZER as ResponseNormalizer<Record>
+  }
+
+  override fun <R> readTransaction(transaction: Transaction<ReadableStore, R>): R {
+    return transaction.execute(this)!!
+  }
+
+  override fun <R> writeTransaction(transaction: Transaction<WriteableStore, R>): R {
+    return transaction.execute(this)!!
+  }
+
+  override fun normalizedCache(): NormalizedCache {
+    error("Cannot get normalizedCache: no cache configured")
+  }
+
+  override fun cacheKeyResolver(): CacheKeyResolver {
+    error("Cannot get cacheKeyResolver: no cache configured")
+  }
+
+  override fun <D : Operation.Data, T, V : Operation.Variables> read(
+      operation: Operation<D, T, V>): ApolloStoreOperation<T> {
+    error("Cannot read operation: no cache configured")
+  }
+
+  override fun <D : Operation.Data, T, V : Operation.Variables> read(
+      operation: Operation<D, T, V>, responseFieldMapper: ResponseFieldMapper<D>,
+      responseNormalizer: ResponseNormalizer<Record>, cacheHeaders: CacheHeaders): ApolloStoreOperation<Response<T>> {
+    error("Cannot read operation: no cache configured")
+  }
+
+  override fun <F : GraphqlFragment> read(fieldMapper: ResponseFieldMapper<F>,
+                                           cacheKey: CacheKey, variables: Operation.Variables): ApolloStoreOperation<F> {
+    error("Cannot read fragment: no cache configured")
+  }
+
+  override fun <D : Operation.Data, T, V : Operation.Variables> write(
+      operation: Operation<D, T, V>, operationData: D): ApolloStoreOperation<Set<String>> {
+    error("Cannot write operation: no cache configured")
+  }
+
+  override fun <D : Operation.Data, T, V : Operation.Variables> writeAndPublish(
+      operation: Operation<D, T, V>, operationData: D): ApolloStoreOperation<Boolean> {
+    error("Cannot writeAndPublish operation: no cache configured")
+  }
+
+  override fun write(fragment: GraphqlFragment, cacheKey: CacheKey,
+                     variables: Operation.Variables): ApolloStoreOperation<Set<String>> {
+    error("Cannot write fragment: no cache configured")
+  }
+
+  override fun writeAndPublish(fragment: GraphqlFragment, cacheKey: CacheKey,
+                               variables: Operation.Variables): ApolloStoreOperation<Boolean> {
+    error("Cannot writeAndPublish fragment: no cache configured")
+  }
+
+  override fun <D : Operation.Data, T, V : Operation.Variables> writeOptimisticUpdates(operation: Operation<D, T, V>, operationData: D, mutationId: UUID): ApolloStoreOperation<Set<String>> {
+    return emptyOperation(emptySet())
```

Comment:
For the optimistic methods, I kept `emptyOperation` as I was unsure if they could be called.